lib/insert/index.js

```diff
@@ -0,0 +1,7 @@
+import _curry3 from '../_internal/_curry3'
+
+export default _curry3(function insert(idx, val, list) {
+  var a = [].concat(list)
+  a.splice(idx, 0, val)
```

Comment:
`splice` is quite slow, maybe use `for` here?

```js
for (...) {
  if (i === idx) arr.push(val)
  arr.push(list[i])
}
```

Reply:
![image](https://user-images.githubusercontent.com/4208480/37023831-2d5692c4-2138-11e8-9d2d-342b5a24c99f.png)